src/main/java/com/seedfinding/mccore/util/data/SpiralIterator.java

```diff
@@ -36,50 +36,46 @@ public SpiralIterator(T center, T radius, Builder<T> builder) {
 	@Override
 	public @NotNull Iterator<T> iterator() {
 		return new Iterator<T>() {
-			private Integer x = null;
-			private Integer z = null;
+			private int x = center.getX();
+			private int z = center.getZ();
 
 			private float n = 1;
 			private int i = 0;
 			private int j = 0;
 
 			@Override
 			public boolean hasNext() {
-				if (x == null || z == null) {
-					return true;
-				}
 				return x >= lowerBound.getX() && x <= upperBound.getX() && z >= lowerBound.getZ() && z <= upperBound.getZ();
 			}
 
 			@Override
 			public T next() {
-				if (x == null || z == null) {
-					x = center.getX();
-					z = center.getZ();
-					return center;
-				}
-				int floorN = (int) Math.floor(n);
-				if (j < floorN) {
-					switch (i % 4) {
-						case 0: z += step; break;
-						case 1: x += step; break;
-						case 2: z -= step; break;
-						case 3: x -= step; break;
-					}
-					j++;
+				try {
 					return builder.build(x, 0, z);
+				} finally {
```

Comment:
Please just store the return value inside a variable instead of using try-finally. Especially since you are returning inside the finally block.